rzk/src/Language/Rzk/VSCode/Lsp.hs

```diff
@@ -46,7 +84,10 @@ handlers =
           then do
             logDebug "rzk.yaml modified. Clearing module cache"
             resetCacheForAllFiles
-          else resetCacheForFiles modifiedPaths
+          else do
+            cache <- getCachedTypecheckedModules
+            actualModified <- dropWhileM (hasNotChanged cache) modifiedPaths
```

Comment:
Currently this makes the (first) changed module to typecheck twice.
Instead of using `Bool`, this should be rewriten in a more constructive style. Can be done separately.